sklearn/cluster/_birch.py

```diff
@@ -522,13 +526,20 @@ def fit(self, X, y=None):
         return self._fit(X, partial=False)
 
     def _fit(self, X, partial):
+        if self.copy != "deprecated":
+            warnings.warn(
+                "`copy` was deprecated in 1.6 and will be removed in 1.8 since it "
+                "has no effect internally. Simply leave this parameter to its default "
+                "value to avoid this warning.",
+                FutureWarning,
+            )
+
         has_root = getattr(self, "root_", None)
         first_call = not (partial and has_root)
```

Comment:
To avoid raising the warning at each call of partial_fit
```suggestion
        has_root = getattr(self, "root_", None)
        first_call = not (partial and has_root)
        
        if self.copy != "deprecated" and first_call:
            warnings.warn(
                "`copy` was deprecated in 1.6 and will be removed in 1.8 since it "
                "has no effect internally. Simply leave this parameter to its default "
                "value to avoid this warning.",
                FutureWarning,
            )
```

Reply:
Sorry overlooked that